cpp/src/io/json/reader_impl.cu

```diff
@@ -111,15 +72,176 @@ constexpr size_t calculate_max_row_size(int num_columns = 0) noexcept
 }  // anonymous namespace
 
 /**
- * @brief Ingest input JSON file/buffer, without decompression
+ * @brief Aggregate the table containing keys info by their hashes.
+ *
+ * @param[in] info Table with columns containing key offsets, lengths and hashes, respectively
+ *
+ * @returns std::unique_ptr<table> Table with data aggregated by hashes
+ */
+std::unique_ptr<table> aggregate_keys_info(std::unique_ptr<table> info)
+{
+  auto const info_view = info->view();
+  std::vector<groupby::aggregation_request> requests;
+  requests.emplace_back(groupby::aggregation_request());
+  requests[0].values = info_view.column(0);  // offsets
+  requests[0].aggregations.push_back(make_min_aggregation());
+  requests[0].aggregations.push_back(make_nth_element_aggregation(0));
```

Comment:
Not sure if this is possible but it would be amazing if a ctor could be added to `groupby::aggregation_request` so we could end up with a single line of code:
```cpp
std::vector<groupby::aggregation_request> requests{
   groupby::aggregation_request{
      info_view.column(0), 
      {make_min_aggregation(), make_nth_element_aggregation(0)}};
```
same for below (could potentially even be combined with `initializer_list` after the fact.

Reply:
Doesn't look like this is possible, the initializer list ctor always copies the elements :(

Reply:
mmm kk, makes sense.